### ISSUE TITLE
Dodo Code Discord Channel retrieval issue

### ISSUE THREAD

**Issue by Numeritxs** [berichan/SysBot.ACNHOrders#15]
Hi! My name is Iván and I think I found an issue in the "EchoDodoChannels" function of the bot.

When the island crashes and the bot auto retrieves new code, it automatically saves it to a text file you located in the "DodoRestoreFilename" config instruction, but if you have the EchoDodoChannels instruction active aswell, the dodo code isn't posted to the channel until someone joins your island.
I think the Dodo should be posted into the channel at the same time it is save in the Dodo text file.

I've made some screenshots to show you the issue:
![2](https://user-images.githubusercontent.com/72006751/141662094-08d5680e-278e-400c-b9ff-972921a37dd4.png)
![1](https://user-images.githubusercontent.com/72006751/141662093-bba4705a-f456-43f1-9dc1-9c3cfac308f4.png)

**Comment by berichan** [comment]
This looks more like a long wait on the discord thread. Does this happen every time with a restore? There's nothing in the source that corresponds to a fly-in & update message blocker

**Comment by Numeritxs** [comment]
Yup, it does happen every time, even in the first dodo code retrieval, it looks like the if from line 176 in Bot/CrossBot.cs isn't returning true and isn't executing, it may be the DodoPosition.IsDodoValid(DodoCode) function that is only used in that if? 
I'm going to sleep since it is 1:30 AM but tomorrow morning i'll be doing some research and post it here if you want ^^

**Comment by Numeritxs** [comment]
Oh i haven't read all your comment, you're right, there's nothing that syncs fly-in and dodo retrieval message lol, well, i'm going to test it further tomorrow!

**Comment by berichan** [comment]
> line 176 in Bot/CrossBot.cs

> it may be the DodoPosition.IsDodoValid(DodoCode) function

This is just a Regex.Match, the thread cannot block here.

**Comment by Numeritxs** [comment]
I've been doing some test and the only thing I figured out is that sometimes the if function in line 176 isn't executing (idk why) so i've used a workaround that was copy and paste lines 176 and 177 to below 314. 
I know that it is copying and pasting the same information (making redundancy) but it is the only way in which I have made it work, I also do not know why sometimes it does not execute the if function of line 176 (I wrote some LogUtil.LogInfo with information from the IF and was returning true), so I guess I need to dig deeper.

Also I want to ask you if there's any way to execute this bot with Yuzu or any emulator, as if I want to test the app I need to turn off the ordering bot from my discord. Tysm!
